src/lib.rs

```diff
@@ -1250,6 +1250,20 @@ where
 {
 }
 
+impl<K, V, S> Index<K> for FlurryHashMap<K, V, S>
+where
+    K: Sync + Send + Clone + Eq + Hash,
+    V: Sync + Send,
+    S: BuildHasher,
+{
+    type Output = V;
+
+    fn index(&self, idx: K) -> &Self::Output {
+        let guard = epoch::pin();
+        self.get(&idx, &guard).expect("no entry found for key")
```

Comment:
This line fails to compile with `cannot return value referencing local variable 'guard'`, but I have no idea how I can fix this.

Reply:
Ah, yes, the return value from `get` is tied to the lifetime of the guard. The guard is what ensures that the reference we return does not get garbage collected while we still have a reference to it! Since you construct the guard inside the function, it gets dropped when the function returns, but you are trying to return the reference nonetheless.

This is definitely a little annoying. The way we get around this for our other methods is that we take the guard as _input_ to the function, but that won't quite work here due to the trait forcing a particular method signature. I wonder if you could do:
```rust
impl<'a, K, V, S> Index for (&'a FlurryHashMap<K, V, S>, &'a Guard) {
```
It's not super pretty, but I _think_ it would let people write code like:
```rust
let map = ...;
let guard = epoch::pin();
let v = &(&map, &guard)[key];
```
Although on second thought, maybe not, since the lifetime of the returned reference would be tied to the lifetime of the tuple rather than the `'a`... Not sure how to go about this actually. Maybe try it and see what happens? You could try
```rust
type Output = &'a V;
```
In which case the return value would be of type `&&'a V`. It's a little awkward, but not _terrible_.

I think the best path forward is to try the above and then see what _using_ the `Index` trait feels like. If it is too cumbersome we can just leave it out.

Reply:
Okay I will try it, thanks